lib/plugins/TaskRunner/LAMPApp.js

```diff
@@ -211,13 +211,18 @@ class LAMPApp extends Script {
       };
       for (let key in paths) {
         let path = paths[key];
-        if (options[key] != undefined) {
+        if (options[key] != undefined && !this.isEmptyObject(options[key])) {
           this.addPhpOptionsToPath(path, options[key]);
-          delete options[key];
+          if (key != 'cli') {
+            this.options.restartApache = true;
+          }
         }
+        delete options[key];
+      }
+      if (!this.isEmptyObject(options)) {
```

Comment:
Line 222 will always be true because line 44 sets options.all ={} even if no options were set. I think we can remove line 44

Reply:
Line 44 is needed because there's a chance that an option will be added to it in addScriptPhpConstants().  It will throw an error if it doesn't exist.  We could check for existence at that point and add it but I thought that this would be better for future code in case we add more options to it.  That way we don't always have to check it.

Also, line 222 won't always be true because just above in line 220, the 'all' key is cleared out.  So, options is only populated if someone adds a php option in the old manner that is outside of 'all'.

Reply:
Thanks, i added a comment in there since i got confused there